### ISSUE TITLE
regression: scalameta(-2) started failing

### ISSUE THREAD

**Issue by SethTisue** [scala/community-builds#663]
last green run:
* https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/2514/consoleFull
* Scala version: 2.12.5-bin-8b65cb2 (scala/scala@8b65cb2)

first red run:
* https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/2516/consoleFull
* Scala version: 2.12.5-bin-9b4554d (scala/scala@9b4554d)

in both runs we are using https://github.com/scalacommunitybuild/scalameta/commits/community-build-2.12 which is frozen

```
[scalameta-2] Processing subproject: semanticdbScalac
...
[scalameta-2] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.9/project-builds/scalameta-2-ef0c8ebfc14bf08461b9ab8f945788e69c0fd3f3/scalameta/semanticdb-scalac/src/main/scala-2.12.4/scala/meta/internal/SemanticdbAnalyzer.scala:1024: value samTp is not a member of Boolean
[scalameta-2] [error]                 if (samAttach.samTp ne NoType) tree.setType(samAttach.samTp).updateAttachment(samAttach)
[scalameta-2] [error]                               ^
[scalameta-2] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.9/project-builds/scalameta-2-ef0c8ebfc14bf08461b9ab8f945788e69c0fd3f3/scalameta/semanticdb-scalac/src/main/scala-2.12.4/scala/meta/internal/SemanticdbAnalyzer.scala:1024: value samTp is not a member of Boolean
[scalameta-2] [error]                 if (samAttach.samTp ne NoType) tree.setType(samAttach.samTp).updateAttachment(samAttach)
[scalameta-2] [error]                                                                       ^
[scalameta-2] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.9/project-builds/scalameta-2-ef0c8ebfc14bf08461b9ab8f945788e69c0fd3f3/scalameta/semanticdb-scalac/src/main/scala-2.12.4/scala/meta/internal/SemanticdbAnalyzer.scala:1024: value samTp is not a member of Boolean
[scalameta-2] [error]                 if (samAttach.samTp ne NoType) tree.setType(samAttach.samTp).updateAttachment(samAttach)
[scalameta-2] [error]                               ^
[scalameta-2] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.9/project-builds/scalameta-2-ef0c8ebfc14bf08461b9ab8f945788e69c0fd3f3/scalameta/semanticdb-scalac/src/main/scala-2.12.4/scala/meta/internal/SemanticdbAnalyzer.scala:1024: value samTp is not a member of Boolean
[scalameta-2] [error]                 if (samAttach.samTp ne NoType) tree.setType(samAttach.samTp).updateAttachment(samAttach)
[scalameta-2] [error]                                                                       ^
[scalameta-2] [error] two errors found
[scalameta-2] [info] No documentation generated with unsuccessful compiler run
[scalameta-2] [error] two errors found
```

`git hist --merges 9b4554d ^8b65cb2` shows the following Scala PRs were merged: scala/scala#6087, scala/scala#6176, scala/scala#6207, scala/scala#6217, scala/scala#6238, scala/scala#6159, scala/scala#6247, scala/scala#6251, scala/scala#6231, scala/scala#6259

**Comment by SethTisue** [comment]
okay the relevant PR is @hrhino's scala/scala#6087:

```diff
-    def inferSamType(fun: Tree, pt: Type, mode: Mode): SAMFunction = {
+    def inferSamType(fun: Tree, pt: Type, mode: Mode): Boolean = fun match {
```

hence "value samTp is not a member of Boolean"

**Comment by SethTisue** [comment]
the problem is reproducible outside of the community build with:

```
hub clone scalameta/scalameta
cd scalameta
hub remote add scalacommunitybuild
git fetch scalacommunitybuild
git checkout community-build-2.12
```

then edit build.sbt as follows:

```diff
-  addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.full),
+  addCompilerPlugin("org.scalamacros" % "paradise_2.12.4" % "2.1.0"),
```

then launch sbt, then:

```
++2.12.5-bin-9b4554d
set unmanagedSourceDirectories in (semanticdbScalac, Compile) += (baseDirectory in semanticdbScalac).value / "src" / "main" / "scala-2.12.4"
semanticdbScalac/compile
```

and boom.

**Comment by SethTisue** [comment]
@hrhino @olafurpg how shall we proceed here? @hrhino does this look fixable to you by adding a `scalameta/semanticdb-scalac/src/main/scala-2.12.5` directory with a tweaked version of the 2.12.4 code? or will your scala/scala#6087 changes need to be adjusted to re-expose what scalameta is using?

cc @adriaanm

**Comment by hrhino** [comment]
@SethTisue I'm checking right now.

As I understand it, they copy the relevant parts of `SemanticdbAnalyzer` from version to version, so someone's going to have to make a 2.12.5 PR. If it's not too early I'd be glad to do that for scalameta (ping @olafurpg) and sit on it until the release, assuming that this doesn't complicate things too much.

Otherwise I suspect a tweak to the compiler to re-expose an API like this wouldn't be too much to do, either.

Sorry about that D:

**Comment by SethTisue** [comment]
> I'd be glad to do that for scalameta (ping @olafurpg) and sit on it until the release, assuming that this doesn't complicate things too much

sgtm. as soon as you've PRed it, I can pull the changes into https://github.com/scalacommunitybuild/scalameta/commits/community-build-2.12 in order to get the community build green again.

**Comment by hrhino** [comment]
So I lied; I don't think I should actually make a 2.12.5 PR until the 2.12.5 release is finalized, since it involves going into `Namers` and `Typers` and grabbing all of the matching changes, so it would indeed be premature to take a stab at that.

I did make a PR (scalacommunitybuild/scalameta#3) which ~will~ ought to unbreak the community build: it simply adapts changes so far in those two files.

**Comment by SethTisue** [comment]
> scalacommunitybuild/scalameta#3

(continued discussion there)

**Comment by SethTisue** [comment]
fixed in c5c11cc3f90088f09e1a8ef01fa4bb418fef78dd
